lianad/src/payjoin/receiver.rs

```diff
@@ -317,47 +320,44 @@ fn send_payjoin_proposal(
 
 fn process_receiver_session(
     db: &sync::Arc<sync::Mutex<dyn DatabaseInterface>>,
+    session_id: SessionId,
     bit: &mut sync::Arc<sync::Mutex<dyn BitcoinInterface>>,
     desc: &descriptors::LianaDescriptor,
     secp: &secp256k1::Secp256k1<secp256k1::VerifyOnly>,
 ) -> Result<(), Box<dyn Error>> {
     let mut db_conn = db.connection();
-    for session_id in db_conn.get_all_active_receiver_session_ids() {
-        let persister = ReceiverPersister::from_id(Arc::new(db.clone()), session_id.clone());
+    let persister = ReceiverPersister::from_id(Arc::new(db.clone()), session_id.clone());
```

Comment:
We had talked about moving `db` up and just passing the `db_conn` since it locks so perhaps we could get into trouble locking it above in `payjoin_receiver_check`
```
impl DatabaseInterface for sync::Arc<sync::Mutex<dyn DatabaseInterface>> {
    fn connection(&self) -> Box<dyn DatabaseConnection> {
        self.lock().unwrap().connection()
    }
}
```
However, we access the entire `DatabaseInterface` in addition to the `DatabaseConnection` in this function since we need to get the persister for each session_id, perhaps also pass down the persister for each session?

Reply:
yeah that would make more sense

Reply:
I am currently passing both db_conn and persister down since the current way we construct the persister is from a `DatabaseInterface` via db not db_conn, I just construct the persister within the loop.